### DISCUSSION TITLE
please make more tutorial about using sokol for 2d games, apps

### DISCUSSION THREAD

**Discussion by timi5691** [vlang/v#18143] [General]
sokol showcases very impressed to me, but i can't understand the examples. please make more simple and easy to understand examples, tutorials.
How to draw shape, images to screen? how to handle keyboard, mouse, touch screen? how to draw to a viewport? how to draw text? how to measure text size? how to draw text with a font?How to create mobile games for different screen sizes and resolutions?

**Comment by Wajinn** [comment]
V's usage is a bit different, because it has [gg](https://modules.vlang.io/gg.html) and [gx](https://modules.vlang.io/gx.html).  The purpose (as I interpret it) seems to be to have a V way of doing simple graphics, which is presently implemented using sokol, but leaves the door open for a completely V implementation or use of some other 3rd party library in the future.

Thus it appears to be better to look at the [V examples](https://github.com/vlang/V/tree/master/examples).   There are some other V games, that for some reason are not on the [awesome V](https://github.com/vlang/awesome-v) list of [games](https://github.com/vlang/awesome-v#games), that you might want to look at too.    

[Dice-dash](https://github.com/Lassi-Koykka/dice-dash)
Note- below ones need updating, but can be used as references
[Vasteroids](https://github.com/islonely/vasteroids)
[Vchess](https://github.com/hedgeg0d/vchess)
[Game of Life](https://github.com/Niklas212/Game-of-Life) 

By the way, your [Gridpathfinding](https://github.com/timi5691/gridpathfinding), should be on awesome V.

There is also an alternative UI toolkit for V, called [MUI](https://github.com/malisipi/mui), which might help you do some of the things you want.  To put them on Android, you would use [VAB](https://github.com/vlang/vab).  Probably somebody more into games or one of the developers should chime in, to give a more comprehensive view of V's relationship to sokol or more detailed answers to your questions.

**Reply by JalonSolov** [reply]
There also games in the examples, such as `examples/2048`.
